torchaudio/functional/filtering.py

```diff
@@ -938,20 +941,31 @@ def lfilter(
 
     Args:
         waveform (Tensor): audio waveform of dimension of ``(..., time)``.  Must be normalized to -1 to 1.
-        a_coeffs (Tensor): denominator coefficients of difference equation of dimension of ``(n_order + 1)``.
+        a_coeffs (Tensor): denominator coefficients of difference equation of dimension of ``(*, n_order + 1)``.
+                                Where * is the optional number of filter banks.
                                 Lower delays coefficients are first, e.g. ``[a0, a1, a2, ...]``.
                                 Must be same size as b_coeffs (pad with 0's as necessary).
-        b_coeffs (Tensor): numerator coefficients of difference equation of dimension of ``(n_order + 1)``.
-                                 Lower delays coefficients are first, e.g. ``[b0, b1, b2, ...]``.
-                                 Must be same size as a_coeffs (pad with 0's as necessary).
+        b_coeffs (Tensor): numerator coefficients of difference equation of dimension of ``(*, n_order + 1)``.
+                                Where * is the optional number of filter banks.
+                                Lower delays coefficients are first, e.g. ``[b0, b1, b2, ...]``.
+                                Must be same size as a_coeffs (pad with 0's as necessary).
         clamp (bool, optional): If ``True``, clamp the output signal to be in the range [-1, 1] (Default: ``True``)
 
     Returns:
-        Tensor: Waveform with dimension of ``(..., time)``.
+        Tensor: Waveform with dimension of ``(..., *, time)``.
```

Comment:
I think that, things about shape and filter bank (noted as `*` in the current form) needs more clarification.
Specifically, the distinction between the cases of 1D coeffs and 2D coeffs should be explained.
(even though I think the use of `*` here is very sophisticated.)

Reply:
How about
```suggestion
        Tensor: Waveform with dimension of either ``(..., *, time)`` if ``a_coeffs`` and ``b_coeffs`` are 2D Tensors, 
        or ``(..., time)`` otherwise.
```

Reply:
This looks good, but can we simply replace the `*` with something like `num_filters`?

(And in the docstring of `a_coeffs`, `b_coeffs`, can we explicitly say `either 1D with shape (num_order + 1) or 2D with shape (num_filters, num_order + 1)`?)

The thing is, ellipsis is used in many docstrings, but asterisk is very unique to this function, so I think that explicitly mentioning the difference of 1D and 2D are easier to understand.